drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+ * @param bytes_nb	 - Number of bytes to be read in burst.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_get_fifo_value(struct adxl367_dev *dev,
+			       uint8_t* read_data,
+			       uint16_t  bytes_nb)
+{
+	int32_t ret;
+	//511 bytes max FIFO size + 1 cmd byte
+	uint8_t buffer[bytes_nb + 1];
+	uint16_t index;
+
+	memset(buffer, 0, bytes_nb + 1);
+
+	if (dev->comm_type == ADXL367_SPI_COMM) {
+		buffer[0] = ADXL367_READ_FIFO;
+		ret = no_os_spi_write_and_read(dev->spi_desc, buffer, bytes_nb + 1);
+		if (ret)
+			return ret;
+		for (index = 0; index < bytes_nb; index++)
+			read_data[index] = buffer[index + 1];
+	} else {
+		buffer[0] = dev->i2c_slave_address + ADXL367_I2C_READ;
+		buffer[1] = ADXL367_REG_I2C_FIFO_DATA;
+		ret = no_os_i2c_write(dev->i2c_desc, buffer, 2, 0);
+		if (ret)
+			return ret;
+		ret = no_os_i2c_read(dev->i2c_desc, buffer, bytes_nb + 1, 1);
+		if (ret)
+			return ret;
+		for (index = 0; index < bytes_nb; index++)
+			read_data[index] = buffer[index + 1];
+	}
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Performs a masked write to a register.
+ *
+ * @param dev      - The device structure.
+ * @param reg_addr - The register address.
+ * @param data     - The register data.
+ * @param mask     - The mask.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_reg_write_msk(struct adxl367_dev *dev,
+			      uint8_t reg_addr,
+			      uint8_t data,
+			      uint8_t mask)
+{
+	int32_t ret;
+	uint8_t reg_data;
+
+	ret = adxl367_get_register_value(dev, &reg_data, reg_addr, 1);
+	if (ret)
+		return ret;
+	reg_data &= ~mask;
+	reg_data |= data;
+	return adxl367_set_register_value(dev, reg_data, reg_addr);
+}
+
+/***************************************************************************//**
+ * @brief Performs soft-reset.
+ *
+ * @param dev - The device structure.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_software_reset(struct adxl367_dev *dev)
+{
+	int32_t ret;
+	ret = adxl367_set_register_value(dev,
+					 ADXL367_RESET_KEY,
+					 ADXL367_REG_SOFT_RESET);
+	if (ret)
+		return ret;
+
+	// Operating Mode: Standby (reset default).
+	dev->op_mode = ADXL367_OP_STANDBY;
+	// Measurement Range: +/- 2g (reset default).
+	dev->range = ADXL367_2G_RANGE;
+	// ODR: 100Hz (reset default).
+	dev->odr = ADXL367_ODR_100HZ;
+	// FIFO: FIFO disabled (reset default).
+	dev->fifo_mode = ADXL367_FIFO_DISABLED;
+	// FIFO Format: All axis (reset default).
+	dev->fifo_format = ADXL367_FIFO_FORMAT_XYZ;
+	// FIFO Read Mode : 14 bits + CH ID (reset default).
+	dev->fifo_read_mode = ADXL367_14B_CHID;
+
+	no_os_mdelay(20);
```

Comment:
Add a comment please regarding the delay value. Is it from datasheet?

Reply:
Datasheet specifies the default turn on time ~9ms. With 9ms sometimes the chip won't properly start-up so I was told to use 15-20ms delay. (on EZ)

Reply:
I think a comment in the code about the delay usage would be useful.